CMakeLists.txt

```diff
@@ -9,6 +9,8 @@
 cmake_minimum_required(VERSION 3.1)
 project(xtl)
 
+enable_testing()
```

Comment:
I'm not familiar with the CTest stuff, do we have to add `enable_testing` in both `CMakeLists.txt` ?

Reply:
Normally you are only supposed to call the `enable_testing()` function/macro once in your main `CMakeLists.txt`. However, since we allow the test subproject to be build as a standalone with `xtl` as a dependency, I added `enable_testing()` in `test/CMakeLists.txt`. Note that it is guarded by 
```cmake
 if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR) 
```
and `CMAKE_SOURCE_DIR` is the root CMake source dir, so we should be fine.

In short: If you don't add `enable_testing()` in `test/CMakeLists.txt` you aren't able to run the tests with ctest when you are building the tests from the `test` subdirectory.

Reply:
ok, many thanks for the explanation!